mongo/search_index_prose_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/internal/assert"
+	"go.mongodb.org/mongo-driver/internal/require"
+	"go.mongodb.org/mongo-driver/internal/uuid"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func newTestCollection(t *testing.T, client *Client) *Collection {
+	t.Helper()
+
+	id, err := uuid.New()
+	require.NoError(t, err)
+	collName := fmt.Sprintf("col%s", id.String())
+	return client.Database("test").Collection(collName)
+}
```

Comment:
This seems to duplicate logic from `mtest` that automatically creates test collections. Is there a reason we can't use `mtest` for these tests?

Reply:
I've moved it to `mongo\integration`.